build-tools/packages/build-tools/src/common/gitRepo.ts

```diff
@@ -208,21 +208,39 @@ export class GitRepo {
 	 */
 	public async getModifiedFiles(): Promise<string[]> {
```

Comment:
would be nice to have a unit test covering the case being fixed here, especially given that we thought it was fixed before.

If we factor the data parsing apary from the async git interactions, it should be easy to test, and we can use the known repro to generate the test case.

Reply:
agreed and I think adding coverage is in the backlog.

Reply:
It should be trivial to just add a regression test for this fix as part of this change. It would be much easier to review with such a test confirming its works. Coverage for the other cases can wait.